src/expr/src/expr/expr_input_ref.rs

```diff
@@ -54,6 +54,12 @@ impl Expression for InputRefExpression {
         }
     }
 
+    fn eval_v2<'a>(&self, input: &'a DataChunk) -> Result<MarkedArrayRef<'a>> {
+        let arr = input.column_at(self.idx).array_ref();
+        let vis = input.vis();
+        Ok(MarkedArrayRef::from_ref(arr, vis))
+    }
```

Comment:
It seems that the only usage of `MarkedArrayRef` is to preserve the `vis` array in `input: &DataChunk`, so why not just let `eval()` be **unaware** of `vis`? 

```rust
fn eval_v2<'a>(&self, input: &[ArrayImpl]) -> Result<ArrayImpl> {
```

Reply:
It will be wasteful, aware of vis is necessary to skip some heavy computation.

Reply:
Yes, my mistake. Then how about

```
fn eval_v2<'a>(&self, input: &[ArrayImpl], vis: &Vis) -> Result<ArrayImpl> {
```

Reply:
It’s almost same as current eval.

Of course I know it work, but I think the interface is unsafe. We should not guarantee what will the output value be if the input vis = false. I want to investigate some safe interface.